unpack-trees.c

```diff
@@ -1454,9 +1475,10 @@ static int clear_ce_flags(struct index_state *istate,
 				istate->cache_nr,
 				&prefix,
 				select_mask, clear_mask,
-				pl, 0);
+				pl, 0, 0);
```

Comment:
I noticed that `progress` was added to `index_state` but `progress_nr` was not.  

Should they be kept together (either both in `index_state` or both as `parameters`)?  I'm not sure how useful `progress` is without having the corresponding `progress_nr` (and vice versa).

Reply:
I had not considered putting `progress_nr` into the struct. Now that I have considered it, I still think it should be a method-local variable.

We want to have the progress indicator grow as we move through the recursive algorithm. By keeping it method-local we can use the return value to increase the variable (`progress_nr += processed`, for example). If we put it in the struct, then we need to make sure we _only_ update the value at the tips of the recursion, or the leaves of the tree. This is harder to guarantee correctness, in my opinion.

I should add this consideration into the commit message.